### PR TITLE
Add `extend` option on `has_and_belongs_to_many`.

### PR DESCRIPTION
**Pull request by keepcosmos**
I think `extend` option is missed :smile:

### CODE REVIEW

**Comment by keepcosmos** [comment]
@senny Could you give me feedback? I think it is just missing option.

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+*   Add `extend` option on `has_and_belongs_to_many`.
```
Comment (senny) [inline activerecord/CHANGELOG.md]: I'd reword this to:

> Pass `:extend` option for `has_and_belongs_to_many` associations to the underlying `has_many :through`.

File: activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb [hunk 2]
```diff
@@ -577,6 +577,11 @@ def test_find_in_association_with_options
     assert_equal developers(:poor_jamis), projects(:active_record).developers.where("salary < 10000").first
   end
 
+  test "test_association_with_extend_option" do
+    post = posts(:welcome)
```
Comment (senny) [inline activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb]: instead of expanding the global models, could you test it with an anonymous subclass defined in the test case?

**Comment by senny** [comment]
@keepcosmos added a few minor comments. Also please make sure that the PR only includes a single commit. Currently you also have a merge commit included: "Merge branch 'master' into add-extend-option-on-habtm" (64793c17fd7d7b4cdddde1187d71230a4775f3fe).

File: activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb (hunk 2 above)
Comment (keepcosmos) [inline activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb]: @senny  It is difficult to define an anonymous subclass. `Builder::HasAndBelongsToMany#middle_reflection` requires class name.
So, I just define a class on the test file.

**Comment by keepcosmos** [comment]
@senny I accept your advice. Thanks! :smile:

**Comment by senny** [comment]
@keepcosmos thank you :yellow_heart:
